enhancements/network/static-ip-addresses-vsphere.md

```diff
@@ -0,0 +1,613 @@
+  state: Active
+  strategy: {}
+  template:
+    machineType: machines_v1beta1_machine_openshift_io
+    machines_v1beta1_machine_openshift_io:
+      failureDomains:
+        platform: VSphere
+        vsphere:
+        - name: fd-2
+        - name: fd-3
+        - name: fd-4
+      metadata:
+        labels:
+          machine.openshift.io/cluster-api-cluster: ngirard-dev-bwnz9
+          machine.openshift.io/cluster-api-machine-role: master
+          machine.openshift.io/cluster-api-machine-type: master
+      spec:
+        lifecycleHooks: {}
+        metadata: {}
+        providerSpec:
+          value:
+            apiVersion: machine.openshift.io/v1beta1
+            credentialsSecret:
+              name: vsphere-cloud-credentials
+            diskGiB: 100
+            kind: VSphereMachineProviderSpec
+            memoryMiB: 16384
+            metadata:
+              creationTimestamp: null
+            network:
+              devices:
+              - addressesFromPools:
+                - group: installer.openshift.io
+                  name: default-0
+                  resource: IPPool
+                nameservers:
+                - 8.8.8.8
+            numCPUs: 4
+            numCoresPerSocket: 2
+            snapshot: ""
+            template: ""
+            userDataSecret:
+              name: master-user-data
+            workspace: {}
+status: {}
+~~~
+
+Prior to installation, the cluster admin may modify the generated machines, ControlPlaneMachineSet definition, IPAddresses and IPAddress claims to match how they wish to maintain the IPs after cluster creation. The installer will choose a default IPPool to create that is not owned by any IPAM controller present in the cluster.  This means after cluster creation, there will be nothing maintaining those IPs.
+
+If the CPMS definition is changed after the cluster creation, the CPMS Operator will assume that to be a global change to masters and will roll out new masters to meet requirements of the new definition.  If the changes are related to the AddressesFromPools, new machines will be created with references to that pool.  The IPAM controller owning the pool definition will need to generate new IPAddress custom resources and update the IPAddressClaims to reference these IPAddress custom resources.  Once these conditions are met, the Machine API operator will begin creation of the new masters.
+
+#### Changes Required
+
+##### Installer
+
+1. Modify the `install-config.yaml` vSphere platform specification to support the definition of the 
+~~~go
+// Hosts defines `Host` configurations to be applied to nodes deployed by the installer
+type Hosts []Host
+
+// Host defines host VMs to generate as part of the installation.
+type Host struct {
+  // FailureDomain refers to the name of a FailureDomain as described in https://github.com/openshift/enhancements/blob/master/enhancements/installer/vsphere-ipi-zonal.md
+  // +optional
```

Comment:
cc @vr4manta can you answer this one?

Reply:
If FailureDomain is not specified, the machine generation logic in installer will auto assign it to a FD just as it would for zonal.  If FDs are not configured in the install-config under platform/vsphere, a "generated" FD is created and all nodes will be placed in that FD.  The Hosts field's intention is to allow you to force a placement in a specific FD with the defined static IP.